gpu-operator/dra-driver.rst

```diff
@@ -0,0 +1,504 @@
+
+- A ComputeDomain resource. 
+
+- A ResourceClaimTemplate to use for workload pods.
+
+  This is used to request access to a unique IMEX channel on a set of compute nodes.
+  Each worker of a multi-node job should reference this ResourceClaimTemplate to ensure that they have access to the same IMEX channel inside the appropriate ComputeDomain. 
+
+- ResourceClaimTemplate for the DaemonSet.
+
+  This is used to request access to a device that injects ComputeDomain-specific settings for an imex-daemon to run with (config.cfg, nodes_config.cfg, etc.). 
+  Each worker of the DaemonSet mentioned below references this ResourceClaimTemplate to ensure they start up IMEX daemons associated with the appropriate ComputeDomain.
+
+- A DaemonSet. 
+
+  Depending on your workload requirements, this DaemonSet forms one (or more) IMEX domains by running IMEX daemons on the set of nodes in a ComputeDomain that requires them.
+  When your workload is deployed, these daemons “follow” the workload pods to the nodes where they have been scheduled. 
+  Through DRA, these daemons are guaranteed to be fully up and running before the workload pods that triggered their creation are allowed to run.
+
+As workload pods that reference a CustomDomain ResourceClaimTemplate, get scheduled they trigger the GPU DRA Driver to request access to the same IMEX channel on whatever node they land on. 
+
+Once scheduled to a node, the GPU DRA Driver adds a Node label for the ComputeDomain to the node where the workload has been scheduled to indicate the node is part of that ComputeDomain.
+This label is used as a NodeSelector on the DaemonSet mentioned above to trigger the scheduling of its pods to specific nodes.
+
+The addition of this Node label triggers the DaemonSet to deploy an IMEX daemon to that node and start running it.
+
+When all daemons have been fully started, the GPU DRA Driver unblocks each worker, injects its IMEX channel into the worker and allows it to start running.
+
+Once all workloads running in a ComputeDomain have run to completion, the label gets removed even if the ComputeDomain itself hasn't been deleted yet.
+This allows these nodes to be reused for other ComputeDomains.
+
+Node and Pod Affitnity Strategies
+=================================
+
+A ComputeDomain isn't strictly about IMEX channels—it's about running workloads across a group of compute nodes. 
+This means even if some nodes are not IMEX capable, they can still be part of the same ComputeDomain.
+You must apply NodeAffinity and PodAffinity rules to make sure your workloads run on IMEX capable nodes.
+
+For example you could set PodAffinity with a preferred topologyKey set to ``nvidia.com/gpu.clique`` for workloads to span multiple NVLink domains but want them packed as tightly as possible. Or use a required topologyKey set to ``nvidia.con/gpu.clique`` when you require all workloads deployed into the same NVLink domain, but don't care which one. 

```

Comment:
I want to add my understanding here and then we can think about potentially rewording/reordering, simplifying.

Technically, when _requiring_ the value of  `nvidia.con/gpu.clique`  to be the same among all jobs then this is the same as saying: all jobs must land in the same domain&clique. And that means: all jobs are mutually reachable (all-to-all communication is possible). For a MNNVL setup this is what's typically wanted, and maybe we should lead with that.

So, I consider this use case to be more advanced or exotic:

>  for workloads to span multiple NVLink domains but want them packed as tightly as possible.

But maybe I don't know enough.